sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -209,10 +210,39 @@ def create_node(self, node_config, tags, count):
         start = time.time()
         while True:
             if time.time() - start > self.timeout:
-                raise config.KubernetesError(
+                pod_status = new_nodes[0].status.phase
+                pod_name = new_nodes[0]._metadata._name
+                events = kubernetes.core_api().list_namespaced_event(
+                    self.namespace,
+                    field_selector=f"involvedObject.name={pod_name}")
+                for event in events.items:
+                    if event.reason == 'FailedScheduling':
+                        event_message = event.message
+                        break
+                timeout_err_msg = (
                     'Timed out while waiting for nodes to start. '
                     'Cluster may be out of resources or '
                     'may be too slow to autoscale.')
+                if event_message is not None:
+                    if pod_status == 'Pending':
+                        if 'Insufficient cpu' in event_message:
+                            total_cpus = os.cpu_count()
+                            raise config.KubernetesError(
+                                'More than available CPU(s) are requested. '
+                                f'In total, {total_cpus} CPUs are available in '
+                                'the machine. Run \'sky status\' to see the '
+                                'number of CPUs that are already in use.')
+                        if 'didn\'t match Pod\'s node affinity/selector' in event_message:
+                            node_selector = pod.spec.node_selector
+                            if node_selector is not None:
+                                raise config.KubernetesError(
+                                    'Unavailable GPU(s) are requested. '
+                                    f'Please confirm if {node_selector} is '
+                                    'available in the cluster.')
+                    raise config.KubernetesError(
```

Comment:
When you ran `sky launch` for this example, I'm assuming your machine would have ran out of CPUs as well. The implementation prints out the message for lack of CPUs when it only sees the message `'Insufficient cpu'` from event Message(the `Message` you see from running `kubectl describe pods`). When there is enough CPU and insufficient memory, it would print out:
```
W 08-26 20:45:24 cloud_vm_ray_backend.py:977]   sky.skylet.providers.kubernetes.config.KubernetesError: Timed out while waiting for nodes to start. Cluster may be out of resources or may be too slow to autoscale. For more details: 0/1 nodes are available: 1 Insufficient memory. preemption: 0/1 nodes are available: 1 No preemption victims found for incoming pod..
```

Reply:
I'll add another UX message to handle `'Insufficient memory'` as well.